src/blocks/content-helper/content-helper-provider.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { select } from '@wordpress/data';
+// eslint-disable-next-line import/named
+import { Schema } from '@wordpress/core-data';
+import Post = Schema.Post;
+import Taxonomy = Schema.Taxonomy;
+import User = Schema.User;
+
+/**
+ * Internal dependencies
+ */
+import { SuggestedPost } from './models/suggested-post';
+import { GetTopPostsResult, BuildFetchDataQueryResult } from './models/function-results';
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+
+interface ApiResponse {
+	error?: string;
+	data?: SuggestedPost[];
+}
+
+class ContentHelperProvider {
+	static async getTopPosts(): Promise<GetTopPostsResult> {
+		const editor = select( 'core/editor' );
+
+		// Get post's author.
+		const currentPost = editor.getCurrentPost() as Post;
```

Comment:
Which type is being overridden here?

Reply:
The type returned is a `Record<string, any>`. In that specific case, using `as` simplifies the code and allows to get correct Intellisense quickly. I would go for keeping for the time being as it also should not have side effects as per the [docs](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions):

> Like a type annotation, type assertions are removed by the compiler and won’t affect the runtime behavior of your code.

Having said that, I do agree that manipulating the original type would be the best practice in general.

Reply:
Sounds good, we can revisit as we get more familiar with Gutenberg dev. :)